### ISSUE TITLE
Regression in 1.7.7 leading to false positive of Credo.Check.Readability.FunctionNames 

### ISSUE THREAD

**Issue by marcelotto** [rrrene/credo#1146]
### Environment

* Credo version (`mix credo -v`): 1.7.7
* Erlang/Elixir version (`elixir -v`): Elixir 1.17.1 (compiled with Erlang/OTP 26)
* Operating system: macOS Sonoma 14.5

### What were you trying to do?

After upgrading to Credo v1.7.7 (from v1.7.5), the following line lead to an issue, which it didn't in the previous version.

```elixir
defmacrop sigil_d(str, _opts) do
```

### Expected outcome

No issue.

### Actual outcome

Code Readability issue:
```
Function/macro/guard names should be written in snake_case
```

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by marcelotto** [comment]
Thanks for the quick fix. 🙏🏻  I can confirm that the problem has been solved. 👍️

**Comment by leobessa** [comment]
Hi @rrrene ! Thanks for the quick fix.

I've opened a PR with some similar fixes: https://github.com/rrrene/credo/pull/1140

**Comment by rrrene** [comment]
@marcelotto @leobessa This is part of Credo `1.7.8`. Thx! 🎉
